### PR TITLE
Cleanup fips provider init

### PR DESCRIPTION
**Pull request by slontis**
Removed dummy evp_test
Changed all algorithm properties to use fips=yes (except for RAND_TEST) (This changes the DRBG and ECX settings)
Removed unused includes.
Added TODO(3.0) for issue(s) that need to be resolved.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: providers/fips/fipsprov.c [hunk 1]
```diff
@@ -727,12 +636,8 @@ int OSSL_provider_init(const OSSL_CORE_HANDLE *handle,
     if (!SELF_TEST_post(&selftest_params, 0))
         goto err;
 
-    /*
-     * TODO(3.0): Remove me. This is just a dummy call to demonstrate making
-     * EVP calls from within the FIPS module.
-     */
-    if (!dummy_evp_call(libctx))
-        goto err;
+    /* TODO(3.0): Tests will hang if this is removed */
```
Comment (mattcaswell) [inline providers/fips/fipsprov.c:639]: Huh?? I wonder why?

File: providers/fips/fipsprov.c [hunk 2]
```diff
@@ -452,80 +361,80 @@ static OSSL_ALGORITHM exported_fips_ciphers[OSSL_NELEM(fips_ciphers)];
...
-    { "DH:dhKeyAgreement", "provider=fips,fips=yes", dh_keyexch_functions },
+    { "DH:dhKeyAgreement", FIPS_DEFAULT_PROPERTIES, dh_keyexch_functions },
 #endif
 #ifndef OPENSSL_NO_EC
-    { "ECDH", "provider=fips,fips=yes", ecdh_keyexch_functions },
-    { "X25519", "provider=fips,fips=no", x25519_keyexch_functions },
-    { "X448", "provider=fips,fips=no", x448_keyexch_functions },
+    { "ECDH", FIPS_DEFAULT_PROPERTIES, ecdh_keyexch_functions },
+    { "X25519", FIPS_DEFAULT_PROPERTIES, x25519_keyexch_functions },
+    { "X448", FIPS_DEFAULT_PROPERTIES, x448_keyexch_functions },
```
Comment (mattcaswell) [inline providers/fips/fipsprov.c:399]: The change for X448/X25519 to approved is deliberate right based on the guidance from the lab?

File: providers/fips/fipsprov.c (hunk 1 above)
Comment (slontis) [reply providers/fips/fipsprov.c:639]: It seems to hang during cleanup with a NULL read lock.. A double free maybe?

File: providers/fips/fipsprov.c (hunk 2 above)
Comment (slontis) [reply providers/fips/fipsprov.c:399]: That was based on the conversation we had during the fips sponsors meeting.

**Comment by slontis** [comment]
Thanks for reviewing. Merged to master

**Pull request closed without merging**